src/py/flwr/client/app.py

```diff
@@ -435,11 +452,43 @@ def _on_backoff(retry_state: RetryState) -> None:
 
                     # Handle app loading and task message
                     try:
-                        # Load ClientApp instance
-                        client_app: ClientApp = load_client_app_fn(fab_id, fab_version)
+                        if isolate and supernode_address is not None:
+                            # Generate SuperNode token
+                            token: int = generate_rand_int_from_bytes(RUN_ID_NUM_BYTES)
```

Comment:
With the change above, this should not be required any more. The case where `isolate` is `True`, but `supernode_address` is unset should be considered an error.

```suggestion
                        if isolate:
                            # Generate SuperNode token
                            token: int = generate_rand_int_from_bytes(RUN_ID_NUM_BYTES)
```

Reply:
i'm afraid but we need to keep the check here. mypy does'tn seem to be smart enough. (but to be fair the change in the other comment happens many lines above, outside two while loops...)

Reply:
got it, resolved